opm/autodiff/AquiferFetkovich.hpp

```diff
@@ -21,367 +21,214 @@ along with OPM.  If not, see <http://www.gnu.org/licenses/>.
-                }
-
-                if (faceDirection == connection.reservoir_face_dir.at(idx))
-                {
-                  faceArea_connected_.at(idx) =  getFaceArea(faceCells, ugrid, faceIdx, idx, connection);
-                  denom_face_areas += ( connection.influx_multiplier.at(idx) * faceArea_connected_.at(idx) );
-                }
-              }
-              auto cellCenter = grid.getCellCenter(cell_idx_.at(idx));
-              cell_depth_.at(idx) = cellCenter[2];
-            }
-
-            const double eps_sqrt = std::sqrt(std::numeric_limits<double>::epsilon());
-            for (size_t idx = 0; idx < cell_idx_.size(); ++idx)
-            {
-              alphai_.at(idx) = (denom_face_areas < eps_sqrt)? // Prevent no connection NaNs due to division by zero
-              0.
-              : ( connection.influx_multiplier.at(idx) * faceArea_connected_.at(idx) )/denom_face_areas;
-            }
+            Base::pa0_ = *(aqufetp_data_.p0);
+          }
+          aquifer_pressure_ = Base::pa0_ ;
+          // use the thermodynamic state of the first active cell as a
+          // reference. there might be better ways to do this...
+          ElementContext elemCtx(Base::ebos_simulator_);
+          auto elemIt = Base::ebos_simulator_.gridView().template begin</*codim=*/0>();
+          elemCtx.updatePrimaryStencil(*elemIt);
+          elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
+          const auto& iq0 = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
+            // Initialize a FluidState object first
+          FluidState fs_aquifer;
+          // We use the temperature of the first cell connected to the aquifer
+          // Here we copy the fluidstate of the first cell, so we do not accidentally mess up the reservoir fs
+          fs_aquifer.assign( iq0.fluidState() );
+          Eval temperature_aq, pa0_mean;
+          temperature_aq = fs_aquifer.temperature(0);
+          pa0_mean = Base::pa0_;
+          Eval mu_w_aquifer = FluidSystem::waterPvt().viscosity(pvttableIdx, temperature_aq, pa0_mean);
+          Base::mu_w_ = mu_w_aquifer.value();
           }
 
-          inline void calculateAquiferCondition()
-          {
-            int pvttableIdx = aqufetp_data_.pvttableID - 1;
-            rhow_.resize(cell_idx_.size(),0.);
-            if (!aqufetp_data_.p0)
-            {
-              pa0_ = calculateReservoirEquilibrium();
-            }
-            else
-            {
-              pa0_ = *(aqufetp_data_.p0);
-            }
-            aquifer_pressure_ = pa0_ ;
-            // use the thermodynamic state of the first active cell as a
-            // reference. there might be better ways to do this...
-            ElementContext elemCtx(ebos_simulator_);
-            auto elemIt = ebos_simulator_.gridView().template begin</*codim=*/0>();
-            elemCtx.updatePrimaryStencil(*elemIt);
+        inline Scalar calculateReservoirEquilibrium()

```

Comment:
Can this function also go the base class? It looks the same to me for AquiferFetkovich and AquiferFetkovich.  Correct me if I misread it.